packages/eslint-plugin/docs/rules/require-types-exports.md

```diff
@@ -0,0 +1,75 @@
+---
+description: 'Require exporting types that are used in exported functions declarations.'
+---
+
+> 🛑 This file is source code, not the primary documentation location! 🛑
+>
+> See **https://typescript-eslint.io/rules/require-types-exports** for documentation.
+
+When exporting functions from a module, it is recommended to export also all the
```

Comment:
> functions

[Feature] It's not just functions though. It could also be, say, variables:

```ts
// source.ts
interface MyType {
  apple: boolean;
}

const value: MyType = { apple: true };

export default value;
```

```ts
import value from "./source.js";

let myValue; // What type is this?!
myValue = value;
```

Reply:
updated the docs